tests/testlib/wakunode.nim

```diff
@@ -36,7 +36,7 @@ proc defaultTestWakuNodeConf*(): WakuNodeConf =
     maxConnections: 50,
     maxMessageSize: "1024 KiB",
     clusterId: 0.uint32,
-    pubsubTopics: @["/waku/2/rs/1/0"],
+    pubsubTopics: @["/waku/2/rs/0/0"],
```

Comment:
Why this change? Seems unrelated to the original purpose of the PR

Reply:
I think it is a bug, because we are setting `clusterId: 0.uint32` but the pubsub topic we are subscribing to corresponds to cluster 1.

Found this bug now that I added a check in this PR to throw an error if we try to subscribe to a pubsub topic that corresponds to a different shard we're on. So fixed it so the pubsub topic corresponds to cluster 0.